utils/firestore.js

```diff
@@ -1131,38 +1138,44 @@ const removeBag = async (siteCode, requestData) => {
           }
       }
 
-      let bagConceptIDIndex = 0;
-      for (let k of Object.keys(box)) { 
-          if (bagConceptIDs.includes(k)) {
-              const currBagConceptID = bagConceptIDs[bagConceptIDIndex];
-              if (currBagConceptID === k) continue;
-              box[currBagConceptID] = box[k];
-              delete box[k];
-              bagConceptIDIndex++;
+      // Create a new sorted box
+      const sortedBox = {};
+      for (let conceptID of bagConceptIDs) {
+          const foundKey = Object.keys(box).find(k => bagConceptIDs.includes(k));
+          if (foundKey) {
+              sortedBox[conceptID] = box[foundKey];
+              delete box[foundKey];
           }
       }
 
+      // Copy any remaining properties from the original box to the sorted box
+      for (let k of Object.keys(box)) {
```

Comment:
Should we use object merge (eg: `sortedBox = {... sortedBox, ... box};`)? It could be a little quicker.

Reply:
Nice. I like this syntax, too. Updated and testing. Thanks.